frontend/lib/resolution/Resolver.cpp

```diff
@@ -2399,7 +2405,23 @@ QualifiedType Resolver::typeForId(const ID& id, bool localGenericToUnknown) {
     ct = inCompositeType;
   } else {
     if (auto rt = methodReceiverType().type()) {
-      if (auto comprt = rt->getCompositeType()) {
+      // get the new class type if the receiver is a class
+      auto nct = rt->toClassType();
+      // get the manager record using ClassType method managerRecordType()
+      if (nct && nct->managerRecordType(context) &&
+          nct->managerRecordType(context)->id() == parentId) {
```

Comment:
All of this can (and, I think, should) be made into a helper function that you can use as:

```C++
auto nct = rt->toClassType();
if (auto mr /* or ct */ = checkIfReceiverIsManagerRecord(context, nct, parentId) {
  auto fieldName =...
}
```